packages/fxa-auth-server/lib/payments/paypal/helper.ts

```diff
@@ -608,4 +610,51 @@ export class PayPalHelper {
       message: 'PayPal refund transaction unsuccessful',
     });
   }
+
+  /**
+   * Attempts to refund all of the invoices passed, provided they're created via PayPal
+   * This will invisibly do nothing if the invoice is not billed through PayPal, so be mindful
+   * if using it elsewhere and need confirmation of a refund.
+   */
+  public async refundInvoices(invoices: Stripe.Invoice[]) {
+    const refundResults = [];
+    const payPalInvoices = invoices.filter((invoice) => {
+      const minCreated = Math.floor(
+        new Date().setDate(new Date().getDate() - MAX_REFUND_DAYS) / 1000
+      );
+      return (
+        invoice.collection_method === 'send_invoice' &&
+        invoice.created > minCreated
+      );
+    });
+    for (const invoice of payPalInvoices) {
+      const transactionId =
+        this.stripeHelper.getInvoicePaypalTransactionId(invoice);
+      if (!transactionId) {
+        this.log.error('PayPalHelper.refundInvoices', {
+          msg: 'Missing transactionId',
+          invoiceId: invoice.id,
+        });
+        continue;
+      }
+      const refundTransactionId =
+        this.stripeHelper.getInvoicePaypalRefundTransactionId(invoice);
+      if (refundTransactionId) {
+        continue;
+      }
+      await this.issueRefund(invoice, transactionId, RefundType.Full);
```

Comment:
Does this ignore errors? We probably want to keep going as much as possible if we encounter PayPal errors, but not if we have connection errors.